vignettes/articles/multi-scale-resampling.Rmd

```diff
@@ -0,0 +1,233 @@
+
+```{r}
+purrr::map_dbl(
+  riemann_resamples$resamples,
+  nrow
+) |> 
+  sum()
+```
+With that said, actually fitting those few thousand models is a two part process. First, we're going to load the rest of the tidymodels packages and use them to define a workflow (from the workflows package), specifying the formula and model that we want to fit to each resample:
+
+```{r message=FALSE}
+library(tidymodels)
+
+lm_workflow <- workflow() |> 
+  add_model(linear_reg()) |> 
+  add_formula(agb ~ n_trees)
+```
+
+Next, we'll actually apply that workflow a few thousand times! We'll calculate two metrics for each run of the model: the root-mean-squared error (RMSE) and the mean absolute error (MAE). We can add these metrics as a new column to our resamples using the following:
+
+```{r message=FALSE}
+riemann_resamples <- riemann_resamples |> 
+  mutate(
+    resampled_outputs = purrr::map(
+      resamples, 
+      fit_resamples,
+      object = lm_workflow,
+      metrics = metric_set(
+        rmse,
+        mae
+      )
+    )
+  )
+```
+
+The `riemann_resamples` object now includes both our original resamples as well as the accuracy metrics associated with each run of the model. A very cool thing about this approach is that we can now visualize our block-level accuracy metrics with a few lines of code. 
+
+For instance, if we wanted to plot block-level RMSE for our largest assessment scale, we could use the following code to "unnest" our nested metric and resample columns:
+
+```{r}
+riemann_resamples$resampled_outputs[[10]] |> 
+  mutate(splits = purrr::map(splits, assessment)) |> 
+  unnest(.metrics) |> 
+  filter(.metric == "rmse") |> 
+  unnest(splits) |> 
+  st_as_sf() |> 
+  ggplot(aes(color = .estimate)) + 
+  geom_sf()
+```
+
+We can also go on to the third step of our assessment process, and get our model accuracy metrics for each aggregation scale we investigated. We'll create a new data frame with only our cellsize variable and the associated model metrics:
+
+```{r}
+riemann_metrics <- riemann_resamples |> 
+  transmute(
+    cellsize = cellsize,
+    resampled_metrics = purrr::map(resampled_outputs, collect_metrics)
+  ) |> 
+  unnest(resampled_metrics)
+
+head(riemann_metrics)
+```
+
+And just like that, we've got a multi-scale assessment of our model's accuracy! We can then use this to investigate and report how well our model does at different levels of aggregation. For instance, by plotting RMSE against MAE at various scales, it appears that our RMSE increases with aggregation while MAE decreases. This hints that, as we aggregate our predictions to larger hexagons, more of our model's overall error is driven by large outliers:
```

Comment:
I think this is the right paragraph to emphasize that we are not tuning cellsize as a hyperparameter but instead thinking about how predictions are used. Maybe say it twice, once very near the beginning briefly and once here with more than one sentence.